lib/src/main/java/com/orange/ods/utilities/composable/KeyboardState.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ *  Copyright 2021 Orange
+ *
+ *  Use of this source code is governed by an MIT-style
+ *  license that can be found in the LICENSE file or at
+ *  https://opensource.org/licenses/MIT.
+ * /
+ */
+
+package com.orange.ods.utilities.composable
+
+import android.graphics.Rect
+import android.view.ViewTreeObserver
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalView
+
+enum class Keyboard {
+    Opened, Closed
+}
+
+/**
+ * Allows to know the soft keyboard state: opened or closed
+ * Add this line anywhere to retrieve the keyboard state: `val keyboardState by keyboardAsState()`
+ */
+@Composable
+fun keyboardAsState(): State<Keyboard> {
+    val keyboardState = remember { mutableStateOf(Keyboard.Closed) }
+    val view = LocalView.current
+    DisposableEffect(view) {
+        val onGlobalListener = ViewTreeObserver.OnGlobalLayoutListener {
+            val rect = Rect()
+            view.getWindowVisibleDisplayFrame(rect)
+            val screenHeight = view.rootView.height
+            val keypadHeight = screenHeight - rect.bottom
+            keyboardState.value = if (keypadHeight > screenHeight * 0.15) {
```

Comment:
Do you remember where does this 0.15 value come from ?

Reply:
I have to confess I pick this on stack overflow 😅